[sensor_ctrl, ast] Preparation for sensor_ctrl d2 (#6291)
lowRISC/opentitan: hw/top_earlgrey/ip/sensor_ctrl/doc/_index.md

---
title: "Sensor Control Technical Specification"
---

# Overview

This document specifies the functionality of the sensor control module.
The sensor control module is a comportable front-end to the [analog sensor top]({{< relref "hw/top_earlgrey/ip/ast/doc" >}}).

It provides basic alert functionality, pad debug hook ups, and a small amount of open source visible status readback.
Long term, this is a module that can be absorbed directly into the `analog sensor top`.

## Features

- Alert hand-shake with `analog sensor top`
- Alert forwarding to `alert handler`
- Status readback for `analog sensor top`
- Pad debug hook up for `analog sensor top`

# Theory of Operations

## Block Diagram

> The diagram below shows how `sensor control` helps `analog sensor top` integration into the overall design.

Reviewer: I assume you will add it later right?

## Alert Handshake

The `analog sensor top` sends alert requests in independent, differential form to the `sensor control`.

> Each alert request is consisted of a pair of signals, one active high and one active low.

Reviewer 2: ```suggestion
Each alert request consists of a pair of signals, one active high and one active low.
```

Author: updated

> However, the active polarity of each signal is independent, this reflects the imprecise timing sensor that drives the alert.

Reviewer 2: nit: two unrelated sentences seem to be joined with a comma

Author: updated

> This means that the `sensor control` recognizes an active alert as long as one of the lines is active, and not the pair of signals being in a particular state.

Reviewer 2: Perhaps say "each signal in the differential pair is a separate dedicated alert signal" to make this clear.

Author: updated

Once an alert request is detected as active, the `sensor control` then formulates a proper alert event through the `prim_alert_sender` and sends a notification to the `alert handler`.

The `sensor control` can optionally generate alert acknowledgements back to the `analog sensor top`.
When an alert request is detected as active, the `sensor control` has 3 options after sending out the alert:

> - Do not acknowledge, thus continually triggering more and more alerts
> - Software acknowledge. This allows software to decide whether an alert should stop triggering.
> - Hardware acknowledge. The hardware immediately acknowledges the alert.

Reviewer 2: ```suggestion
- Do not acknowledge: Thus, let more and more alerts to be continuously triggered.
- Software acknowledge: Let the software to decide whether an alert should stop triggering.
- Hardware acknowledge: Let the hardware immediately acknowledges the alert.
```

Author: updated
